app/cdap/components/hydrator/components/TopPanel/ActionButtons/index.tsx

```diff
@@ -0,0 +1,295 @@
+ * the License.
+ */
+
+import IconSVG from 'components/shared/IconSVG';
+import React from 'react';
+import {
+  ActionButtonsContainer,
+  BorderRightButton,
+  ButtonLabel,
+  CommonButton,
+  CustomTooltip,
+  IconLoading,
+  IconPlay,
+  IconSchedule,
+  IconSliders,
+  IconStop,
+  PreviewModeButton,
+  RunTimeSpan,
+} from './styles';
+
+interface IDuration {
+  minutes: string;
+  seconds: string;
+}
+
+interface IActionButtonsProps {
+  previewMode: boolean;
+  previewEnabled: boolean;
+  togglePreviewMode: () => void;
+  toggleConfig: () => void;
+  viewConfig: boolean;
+  showSchedule: boolean;
+  viewScheduler: boolean;
+  toggleScheduler: () => void;
+  hasNodes: boolean;
+  onSaveDraft: () => void;
+  onPublish: () => void;
+  onImport: () => void;
+  onExport: () => void;
+  onClickLogs: () => void;
+  previewLoading: boolean;
+  previewRunning: boolean;
+  startOrStopPreview: () => void;
+  queueStatus: string;
+  displayDuration: IDuration;
+  loadingLabel: string;
+  currentPreviewId: string;
+  viewLogs: boolean;
+  timerLabel: string;
+}
+
+export const ActionButtons = ({
+  previewMode,
+  previewEnabled,
+  togglePreviewMode,
+  toggleConfig,
+  viewConfig,
+  showSchedule,
+  viewScheduler,
+  toggleScheduler,
+  hasNodes,
+  onSaveDraft,
+  onPublish,
+  onImport,
+  onExport,
+  onClickLogs,
+  previewLoading,
+  previewRunning,
+  startOrStopPreview,
+  queueStatus,
+  displayDuration,
+  loadingLabel,
+  currentPreviewId,
+  viewLogs,
+  timerLabel,
+}: IActionButtonsProps) => {
+  return (
+    <ActionButtonsContainer>
+      {previewMode ? (
+        //   PREVIEW MODE BUTTONS
+        <div>
+          <PreviewModeButton
+            variant="text"
+            onClick={togglePreviewMode}
+            data-cy="preview-active-btn"
+          >
+            <div>
+              <IconSVG name={'icon-eye'}></IconSVG>
+              <ButtonLabel>Preview</ButtonLabel>
+            </div>
+          </PreviewModeButton>
+          <CommonButton
+            active={viewConfig}
+            disabled={previewLoading || previewRunning}
+            variant="text"
+            onClick={!previewLoading && !previewRunning && toggleConfig}
+            data-cy="preview-config-btn"
+          >
+            <div>
+              <IconSliders name={'icon-sliders'}></IconSliders>
+              <ButtonLabel>Configure</ButtonLabel>
+            </div>
+          </CommonButton>
+          {!previewLoading &&
+            (!previewRunning ? (
+              <CustomTooltip
+                title={hasNodes ? '' : 'Start building a pipeline before starting preview'}
+                arrow
+                placement="bottom"
+              >
+                <span>
+                  <CommonButton
+                    disabled={!hasNodes}
+                    variant="text"
+                    onClick={hasNodes && startOrStopPreview}
+                    data-cy="preview-top-run-btn"
```

Comment:
Do the integration tests work with this?

Reply:
That's a great point, running the tests to verify now